apple/internal/apple_framework_import.bzl

```diff
@@ -135,16 +96,55 @@ def _apple_framework_import_impl(ctx):
 
     objc_provider_fields["providers"] = [dep[apple_common.Objc] for dep in ctx.attr.deps]
     objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
-    providers.append(objc_provider)
-    providers.append(
+    return [
+        objc_provider,
         apple_common.new_dynamic_framework_provider(
```

Comment:
this new_dynamic_framework_provider can be moved so that only apple_dynamic_framework_import has it, it's not required for the static one

Reply:
yeah, because it was one rule before